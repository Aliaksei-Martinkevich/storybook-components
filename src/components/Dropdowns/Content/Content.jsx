import React from 'react';
import PropTypes from 'prop-types';
import './Content.scss';

import Item from '../Item/Item.jsx';

const Content = ({ className = '', onItemClick, children }) => {
  const onItemClickHandler = (key) => {
    onItemClick(key);
  };

  return (
    <div className={`dropdown-content ${className}`}>
      {Array.isArray(children)
        ? Array.prototype.map.call(children, (el, id) =>
          (<Item
            onClick={onItemClickHandler.bind(this, el.key || id.toString())}
            key={el.key || id}
          >
            {el}
          </Item>))
        : <Item
          onClick={onItemClickHandler.bind(this, children.key || '0')}
          key={children.key || '0'}
        >
          {children}
        </Item>}
    </div>
  );
};

Content.propTypes = {
  className: PropTypes.string,
  onItemClick: PropTypes.func,
};

export default Content;
