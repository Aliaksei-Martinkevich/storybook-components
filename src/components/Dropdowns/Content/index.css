.dropdown-content {
    position: absolute;
    background-color: white;
    min-width: 160px;
    max-width: 350px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
    z-index: 1;
    border: 0.5px solid #CCC;
}

.collapsed .dropdown-content{
    display: none;
}

.expanded .dropdown-content{
    display: block;
}