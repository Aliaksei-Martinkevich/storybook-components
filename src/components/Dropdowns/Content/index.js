import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

import Item from '../Item';

const Content = ({className = '', onItemClick, children }) => {
  const onItemClickHandler = (itemId) => {
    onItemClick(itemId);
  };

  return (
    <div className={`dropdown-content ${className}`}>
        { Array.isArray(children) 
        ? Array.prototype.map.call(children, (el, id) => 
          <Item onClick={onItemClickHandler} itemId={el.props.itemId || id}>{el}</Item>)
        : <Item onClick={onItemClickHandler} itemId={children.props.itemId || 0}>{children}</Item> }
    </div>
  )
};

Content.propTypes = {
  className: PropTypes.string,
  onItemClick: PropTypes.func, 
};

export default Content;
