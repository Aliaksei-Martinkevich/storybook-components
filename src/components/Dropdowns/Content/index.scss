.dropdown-content {
    position: absolute;
    background-color: white;
    min-width: 160px;
    max-width: 350px;
    z-index: 1;
    border: 0.5px solid #CCC;
}

.collapsed .dropdown-content{
    display: none;
}

.expanded .dropdown-content{
    display: block;
}