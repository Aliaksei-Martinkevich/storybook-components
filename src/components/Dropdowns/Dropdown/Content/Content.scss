.dropdown-content {
    background-color: white;
    z-index: 1;
    border: 0.5px solid #CCC;
    
    &.collapsed {
        display: none;
    }
    &.expanded {
        display: block;
    }
}