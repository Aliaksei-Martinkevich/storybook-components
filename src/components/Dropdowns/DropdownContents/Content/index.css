.dropdown-content {
    position: absolute;
    background-color: white;
    min-width: 160px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
    z-index: 1;
}

.collapsed .dropdown-content{
    display: none;
}

.expanded .dropdown-content{
    display: block;
}