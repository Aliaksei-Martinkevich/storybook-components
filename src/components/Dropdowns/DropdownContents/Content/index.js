import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Content = ({className = '', children }) => {
  return (
    <div className={`dropdown-content ${className}`}>
      <div>
        { Array.isArray(children) 
        ? Array.prototype.map.call(children, el => <div>{el}</div>)
        : <div>{children}</div> }
      </div>
    </div>
  )
};

Content.propTypes = {
  className: PropTypes.string,
};

export default Content;
