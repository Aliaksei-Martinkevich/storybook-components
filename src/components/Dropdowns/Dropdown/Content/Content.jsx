import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './Content.scss';
import Item from './Item/Item.jsx';

const Content = ({ className, onItemClick, children, expanded }) => (
  <div className={classNames('dropdown-content', className, {
    'dropdown-content_expanded': expanded,
    'dropdown-content_collapsed': !expanded })}
  >
    {
        React.Children.map(children,
          (el, id) => (
            <Item
              className="dropdown-content__item"
              onClick={onItemClick}
              key={el.key || id.toString()}
              id={el.key || id.toString()}
            >
              {el}
            </Item>))
    }
  </div>
  );

Content.defaultProps = {
  className: '',
  onItemClick: undefined,
  expanded: false,
};

Content.propTypes = {
  className: PropTypes.string,
  onItemClick: PropTypes.func,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  expanded: PropTypes.bool,
};

export default Content;
