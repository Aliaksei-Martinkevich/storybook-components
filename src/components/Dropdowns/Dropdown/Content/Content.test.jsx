import React from 'react';
import renderer from 'react-test-renderer';
import Content from './Content';

describe('Content', () => {
  it('Should be collapsed', () => {
    const customClassName = 'class-name-example';
    const component = renderer.create(
      <Content
        className={customClassName}
      />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should be expanded', () => {
    const customClassName = 'class-name-example';
    const component = renderer.create(
      <Content
        className={customClassName}
        expanded
      />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Children should be clickable', () => {
    const customClassName = 'class-name-example';
    const onItemClick = jest.fn();
    const component = renderer.create(
      <Content
        className={customClassName}
        onItemClick={onItemClick}
      >
        <div>Child 1</div>
        <div>Child 2</div>
        <div>Child 3</div>
      </Content>);
    let tree = component.toJSON();
    tree.children.forEach(elem => elem.props.onClick());
    tree = component.toJSON();
    expect(onItemClick.mock.calls).toEqual([['0'], ['1'], ['2']]);
    expect(tree).toMatchSnapshot();
  });
});
