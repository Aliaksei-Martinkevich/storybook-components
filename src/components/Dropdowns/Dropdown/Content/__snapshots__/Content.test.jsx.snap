// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Children should be clickable 1`] = `
<div
  className="dropdown-content class-name-example collapsed"
>
  <div
    className="dropdown-content__item"
    onClick={[Function]}
  >
    <div>
      Child 1
    </div>
  </div>
  <div
    className="dropdown-content__item"
    onClick={[Function]}
  >
    <div>
      Child 2
    </div>
  </div>
  <div
    className="dropdown-content__item"
    onClick={[Function]}
  >
    <div>
      Child 3
    </div>
  </div>
</div>
`;

exports[`Content Should be collapsed 1`] = `
<div
  className="dropdown-content class-name-example collapsed"
/>
`;

exports[`Content Should be expanded 1`] = `
<div
  className="dropdown-content class-name-example expanded"
/>
`;
